File: pset1_C/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    // set the value in dollars and check it
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);

    // converting dollars to cents
    int cents = round(dollars * 100);

    // array of available coins
    int coins[] = {25, 10, 5, 1};
    // number of elements in the array
    size_t size = sizeof(coins) / sizeof(coins[0]);

    // the minimum number of coins
    int count = 0;

    for (int i = 0; i < size; i++)
    {
        count += cents / coins[i];
        cents = cents % coins[i];
    }

    printf("%i\n", count);
}

File: pset6_Python/cash/cash.py
from cs50 import get_float
from math import floor


def main():
    # converting dollars to cents
    cents = round(money() * 100)

    # number of coins
    number = 0

    for i in [25, 10, 5, 1]:
        # get the number of coins of each denomination
        number += floor(cents / i)
        # get the number of cents after deducting coins of each denomination
        cents = cents % i

    # print minimum number of coins
    print(number)


# getting a positive number from the user
def money():
    while True:
        dollars = get_float("Change owed: ")
        if dollars > 0:
            break
    return dollars


main()